check-stylish.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -e
+
+function fail_maybe_explain {
+    if [ -n "$TRAVIS" ];
+    then
+        echo "Run ./check-stylish.sh locally to stylize your files / debug this"
+    fi
+    exit 1
+}
+
+if [ -n "$TRAVIS" ];
+then
+    # It takes a while to compile, let's not tax Travis' infrastructure for no
+    # reason. We can't cache it via cabsnap because that's based on the install-
+    # plan for Idris, which of course doesn't have stylish-haskell as a
+    # dependency.
+    STYLISH_EXE=$(mktemp)
+    wget -O $STYLISH_EXE "https://s3-us-west-2.amazonaws.com/idris-travis-binaries/stylish-haskell"
```

Comment:
It's the thing we use to cache dependencies. It puts stuff into Travis' cache.

One reason for the binary is to avoid breaking the build if stylish-haskell breaks for some reason e.g. new versions of its dependencies or a new version of it.

It's possible to freeze the dependencies and cache it, but this was easier. I'll look later tonight. 

The binary's on my S3 account. I agree it'd be better if I weren't solely responsible for maintaining it.